### PR TITLE
BPS-400 Handle NullPointerException when scanned documents are null in ccd case

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-400

### Change description ###
Added assertions to check ccd data and scanned documents are not null

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java [hunk 1]
```diff
@@ -29,9 +30,14 @@ private ScannedDocumentsHelper() {
 
     @SuppressWarnings("unchecked")
     public static List<Document> getDocuments(CaseDetails caseDetails) {
-        List<Map<String, Object>> data = (List<Map<String, Object>>) caseDetails.getData().get("scannedDocuments");
+        Map<String, Object> caseData = caseDetails.getData();
+        Assert.notNull(caseData, "Case Data must not be null");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: where is `getDocuments` used?
Do we actually want to throw `IllegalArgumentException` or return an empty list??

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java [hunk 2]
```diff
@@ -29,9 +30,14 @@ private ScannedDocumentsHelper() {
...
 
-        return data.stream()
+        List<Map<String, Object>> scannedDocuments = (List<Map<String, Object>>) caseData.get("scannedDocuments");
+        // ccd returns an empty list when scanned documents are not present.
+        Assert.notNull(scannedDocuments, "Scanned Documents must not be null");
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: Should we fail in this case? Or maybe we should set it to null?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java (hunk 1 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: I guess it's used for extracting existing documents from case data before adding more. As for the error, maybe we should just treat it as an empty case to add fields to? I don't know myself, just asking.

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: It's being used when attaching evidence to an existing case - `AttachDocsToSupplementaryEvidence.buildCaseDataContent()`. 
we want to throw `IllegalArgumentException` as ccd data is saved as empty collection in database when no scanned documents or case data is not provided

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: After discussing with probate, some user roles (eg: solicitor) don't have access to scannedDocuments field. In such cases, scannedDocuments are null in the ccd case. So I'm changing the code to return empty list

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java (hunk 2 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: I'm changing the code to return an empty list

**Comment by lgonczar** [comment]
👍

**Pull request merged**